src/workflows-app/components/RecordsTable.ts

```diff
@@ -0,0 +1,188 @@
+import _ from 'lodash/fp';
+import { Dispatch, Fragment, SetStateAction, useEffect, useRef, useState } from 'react';
+import { h, span } from 'react-hyperscript-helpers';
+import { AutoSizer } from 'react-virtualized';
+import { Checkbox } from 'src/components/common';
+import { GridTable, HeaderCell, Resizable, Sortable, TextCell } from 'src/components/table';
+import {
+  RecordAttributes,
+  RecordResponse,
+  RecordTypeSchema,
+} from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import colors from 'src/libs/colors';
+import { parseAttributeName } from 'src/workflows-app/utils/submission-utils';
+
+const recordMap = (records: RecordResponse[]) => {
+  return _.fromPairs(_.map((e) => [e.id, e], records));
+};
+
+type RecordsTableProps = {
+  records: RecordResponse[];
+  selectedRecords: Record<string, RecordResponse>;
+  setSelectedRecords: Dispatch<SetStateAction<Record<string, RecordResponse>>>;
+  selectedDataTable: RecordTypeSchema;
+};
+
+const RecordsTable = ({ records, selectedRecords, setSelectedRecords, selectedDataTable }: RecordsTableProps) => {
+  const [dataTableColumnWidths, setDataTableColumnWidths] = useState({});
+  const dataTableRef = useRef<{ recomputeColumnSizes: () => void }>(null);
+  const [recordsTableSort, setRecordsTableSort] = useState<{ field: string; direction: 'asc' | 'desc' }>({
+    field: 'id',
+    direction: 'asc',
+  });
+
+  useEffect(() => {
+    dataTableRef.current?.recomputeColumnSizes();
+  }, [dataTableColumnWidths, records]);
+
+  const selectAll = () => {
+    setSelectedRecords(_.assign(selectedRecords, recordMap(records)));
+  };
+
+  const deselectAll = () => {
+    setSelectedRecords({});
+  };
+
+  const allSelected = () => {
+    const recordIds = _.map('id', records);
+    const selectedIds = _.keys(selectedRecords);
+    return records.length && _.every((k) => _.includes(k, selectedIds), recordIds);
+  };
+
+  const resizeColumn = (currentWidth, delta, columnKey) => {
+    setDataTableColumnWidths(_.set(columnKey, currentWidth + delta));
+  };
+
+  const withDataTableNamePrefix = (columnName) => `${selectedDataTable.name}/${columnName}`;
+
+  const recordsTableData = _.orderBy<RecordResponse & RecordAttributes>(
+    [
+      ({ [recordsTableSort.field]: field }) =>
+        Number.isNaN((field as any) * 1) ? _.lowerCase(field as string) : (field as any) * 1, // converts field to int, float, or NaN (if field is a string)
```

Comment:
Clarifying question - is the reason we need to do this is because of the `RecordResponse & RecordAttributes` type? Or is it something else?

Reply:
Yes, the RecordAttributes type is a map from string to unknown so we need to narrow down the type, and it seems like we had previous logic to account for numbers being stringified so that is maintained by the `* 1` which should convert any strings like `'100'` to `100`